backend/src/ml_space_lambda/app_configuration/lambda_functions.py

```diff
@@ -18,39 +18,234 @@
+    ServiceType.LABELING_JOB: {
+        "Statements": [
+            {
+                IAMStatementProperty.EFFECT: IAMEffect.DENY,
+                IAMStatementProperty.ACTION: [
+                    "sagemaker:CreateLabelingJob",
+                    "sagemaker:DescribeLabelingJob",
+                    "sagemaker:StopLabelingJob",
+                ],
+                IAMStatementProperty.RESOURCE: "*",
+            }
+        ],
+        "ResourceType": ResourceType.LABELING_JOB,
+    },
+    ServiceType.EMR_CLUSTER: {
+        "Statements": [
+            {
+                IAMStatementProperty.EFFECT: IAMEffect.DENY,
+                IAMStatementProperty.ACTION: [
+                    "elasticmapreduce:RunJobFlow",
+                    "elasticmapreduce:ListClusters",
+                    "elasticmapreduce:DescribeCluster",
+                    "elasticmapreduce:ListInstances",
+                    "elasticmapreduce:AddTags",
+                    "elasticmapreduce:TerminateJobFlows",
+                    "elasticmapreduce:SetTerminationProtection",
+                ],
+                IAMStatementProperty.RESOURCE: "*",
+            },
+            {
+                IAMStatementProperty.EFFECT: IAMEffect.DENY,
+                IAMStatementProperty.ACTION: ["iam:PassRole", "iam:ListRoleTags"],
+                IAMStatementProperty.RESOURCE: [
+                    get_account_arn("iam", f"role/{env_variables[EnvVariable.EMR_EC2_ROLE_NAME]}"),
+                    get_account_arn("iam", f"role/{env_variables[EnvVariable.EMR_SERVICE_ROLE_NAME]}"),
+                ],
+            },
+            {
+                IAMStatementProperty.EFFECT: IAMEffect.DENY,
+                IAMStatementProperty.ACTION: ["ec2:AuthorizeSecurityGroupIngress"],
+                IAMStatementProperty.RESOURCE: get_account_arn("ec2", "security-group/*"),
+            },
+            {
+                IAMStatementProperty.EFFECT: IAMEffect.DENY,
+                IAMStatementProperty.ACTION: ["ec2:DescribeInstances", "ec2:DescribeRoutesTables"],
+                IAMStatementProperty.RESOURCE: "*",
+            },
+        ],
+        "ResourceType": ResourceType.EMR_CLUSTER,
+    },
+}
 
-    new_configuration = SettingsModel.from_dict(request["configuration"])
+# Instead of deleting the policy (which requires unattaching from all roles) assign a filler non-impactful statement
+filler_deny_statement = (
+    {
+        IAMStatementProperty.EFFECT: IAMEffect.DENY,
+        IAMStatementProperty.ACTION: ["dynamodb:CreateTable"],
+        IAMStatementProperty.RESOURCE: get_account_arn("dynamodb", "table/invalid-table"),
+    },
+)
 
-    app_configuration = AppConfigurationModel(
-        configScope=configScope,
-        version_id=version_id,
-        configuration=new_configuration,
-        changed_by=event["requestContext"]["authorizer"]["principalId"],
-        change_reason=request["changeReason"],
-        created_at=time.time(),
-    )
 
+@event_wrapper
```

Comment:
The normal @api_wrapper only provides a 200 response with the returned value. @event_wrapper is basically the same, but doesn't assign a 200, so it provides more flexibility for what HTML status code we return

Reply:
Are you sure about that? I thought I was returning a different status code than 200 somewhere else. We definitely don't return 200 on exceptions

Reply:
Yes. [api_wrapper](https://github.com/awslabs/mlspace/blob/main/backend/src/ml_space_lambda/utils/common_functions.py#L100-L113) provides for the binary possibility of a 200 response or a 400 response on exception. It doesn't provide the possibility of other HTTP responses. Since I was doing 207 here, we had to use `event_wrapper`. After other feedback, we may can the `207` since we are pursuing atomic actions, so this could go back to an `api_wrapper`

Reply:
Following up on this one - sounds like we should revert it to an api wrapper

Reply:
In the end we are leaving this as an `event_wrapper` for the following reason:
- Everything good - 200
- Something fails the atomicity - 400+
- Changes applied, but error during resource suspension / cleanup - 207

This gives us bandwidth to handle cases where the `SYSTEM_ROLE` is potentially not handled correctly and helps distinguish it from a complete failure